### ISSUE TITLE
VS Code + notebook setup, Panel + Param, no reactivity when using `pn.template`

### ISSUE THREAD

**Issue by earshinov** [holoviz/panel#6888]
#### Versions

bokeh==3.4.1
ipykernel==6.29.4
jupyter-bokeh==4.0.4
panel==1.4.4
param==2.1.0

#### Expected and observed behavior

```python
import panel as pn
import param

pn.extension()

text = pn.widgets.TextInput()

@param.depends(text.param.value, watch=True)
def _update(value):
    print(value)

text

#text.servable();

#pn.template.VanillaTemplate(main=[text])
```

When rendering `text`, everything is fine.  Typing "1" and moving focus away from the input results in `_update` being called and the value being printed below the input.

![text](https://github.com/holoviz/panel/assets/52021/6ab42b56-4b70-4f13-9dfc-2e74b7c70db3)

When rendering `pn.template` (after commenting out `text` and uncommenting `pn.template.VanillaTemplate(main=[text])`), the `_update` function is never called.

![template](https://github.com/holoviz/panel/assets/52021/62d7324e-fd0a-4c06-9a95-bf506da8207c)

#### Example code

See above or use the repro repo:
https://github.com/earshinov/repro-panel-param-vscode-notebook-pn-template-no-reactivity

#### Screenshots or screencasts of the bug in action

—

#### Notes

A beginner Panel and Param user here. Am I expecting them to support something that they weren't intended to support?

- [ ] I may be interested in making a pull request to address this

**Comment by philippjfr** [comment]
Yes, this configuration is currently not supported. The problem is that templates are meant to be served as a distinct application and in many cases can break the notebook UI. We should definitely handle this case better though.
